core/src/splitting/SplittingRule.h

```diff
@@ -51,6 +51,14 @@ class SplittingRule {
                                std::vector<size_t>& split_vars,
                                std::vector<double>& split_values,
                                std::vector<bool>& send_missing_left) = 0;
+
+ /**
+  * This is an optional public member used for C++ unit testing purposes.
+  *
+  * For example, Survival Splitting sets this to the best logrank statistic
+  * and verifies this is correct in a unit test.
+  */
+  double test_statistic;
```

Comment:
Instead of introducing a mutable variable for testing (which is a bit tricky), it's common to factor out a method that computes the intermediate result. In particular, I wonder if we could add a method like `SurvivalSplittingRule#find_best_split_internal` that returns `best_log_rank`, and check its output in unit tests. Then `SurvivalSplittingRule#find_best_split` would delegate to that method.

Another option is to add `test_statistic` as an output variable to `SplittingRule#find_best_split`. We could just add a comment noting that it's used in unit tests.

Reply:
It sounds like adding it as an output to `SplittingRule#find_best_split` would be the simplest option?

Reply:
Either one works for me, whatever you think turns out cleanest.

Reply:
I added `SurvivalSplittingRule::find_best_split_internal`, that seemed better!